content/en/docs/tasks/administer-cluster/kms-provider.md

```diff
@@ -31,29 +33,53 @@ you have selected.
 {{< feature-state for_k8s_version="v1.12" state="beta" >}}
 
 ### KMS v2
-* Kubernetes version 1.25.0 or later is required
+{{< feature-state for_k8s_version="v1.27" state="beta" >}}
 
-* Set kube-apiserver feature gate: `--feature-gates=KMSv2=true` to configure a KMS v2 provider
+* For version 1.25 and 1.26, enabling the feature via kube-apiserver feature gate is required.
+Set `--feature-gates=KMSv2=true` to configure a KMS v2 provider.
 
 * Your cluster must use etcd v3 or later
 
-{{< feature-state for_k8s_version="v1.25" state="alpha" >}}
+{{< caution >}}
+The KMS v2 API and implementation changed in incompatible ways in-between the alpha release in v1.25
+and the beta release in v1.27.  Attempting to upgrade from old versions with the alpha feature
+enabled will result in data loss.
+{{< /caution >}}
 
 <!-- steps -->
 
 The KMS encryption provider uses an envelope encryption scheme to encrypt data in etcd.
-The data is encrypted using a data encryption key (DEK); a new DEK is generated for each encryption.
+The data is encrypted using a data encryption key (DEK).
 The DEKs are encrypted with a key encryption key (KEK) that is stored and managed in a remote KMS.
-The KMS provider uses gRPC to communicate with a specific KMS plugin.
+With KMS v1, a new DEK is generated for each encryption.
+With KMS v2, a new DEK is generated on server startup and when the KMS plugin informs the API server
+that a KEK rotation has occurred (see `key_id` section below).
+The KMS provider uses gRPC to communicate with a specific KMS plugin over a UNIX domain socket.
 The KMS plugin, which is implemented as a gRPC server and deployed on the same host(s)
 as the Kubernetes control plane, is responsible for all communication with the remote KMS.
 
+{{< caution >}}
+Virtual machine (VM) save state and restore operations are not supported with KMS v2.
+
+The API server uses AES-GCM with a 12 byte nonce (8 byte atomic counter and 4 bytes random data) for encryption. 
+The following issues could occur if the VM is saved and restored:
+1. The counter value may be lost or corrupted if the VM is saved in an inconsistent state or restored improperly. 
+This can lead to a situation where the same counter value is used twice, resulting in the same nonce being used 
+for two different messages.
```

Comment:
Better indent the follow-up lines by 3 spaces for numbered lists.

```suggestion
   This can lead to a situation where the same counter value is used twice, resulting in the same nonce being used 
   for two different messages.
```